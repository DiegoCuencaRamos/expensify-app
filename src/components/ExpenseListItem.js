import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

// load a locale
numeral.register('locale', 'es', {
    delimiters: {
        thousands: ' ',
        decimal: ','
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '€'
    }
});

// switch between locales
numeral.locale('es');

export const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div>
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__subtitle">{numeral(amount / 100).format('0,0[.]00 $')}</span>   
        </div>
        <h3 className="list-item__data">{moment(createdAt).format('MMMM Do, YYYY')}</h3>
    </Link>
);

export default ExpenseListItem;