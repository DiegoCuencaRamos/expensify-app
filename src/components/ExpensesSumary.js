import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSumary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('0,0[.]00 $');

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
};

const mapStateToProps = (state) => {
    console.log('ExpensesSumary state:', state)

    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    } 
};

console.log(connect(mapStateToProps)(ExpensesSumary))

export default connect(mapStateToProps)(ExpensesSumary);

