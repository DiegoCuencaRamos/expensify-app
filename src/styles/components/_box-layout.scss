@use '../base/settings';

.box-layout {
    align-items: center;
    background-color: #444444;
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background-color: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    padding: settings.$l-size settings.$m-size;
    text-align: center;
    width: 25rem,
}

.box-layout__title {
    margin: 0 0 settings.$m-size 0;
    line-height: 1;
}

