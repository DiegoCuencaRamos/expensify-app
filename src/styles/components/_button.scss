@use '../base/settings';

.button {
    color: #ffffff;
    background-color: settings.$blue;
    font-size: settings.$font-size-large;
    font-weight: 300;
    padding: settings.$s-size;
    border: none;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button-secondary {
    background: #888;
}