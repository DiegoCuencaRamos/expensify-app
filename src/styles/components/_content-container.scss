@use '../base/settings';

.content-container {
    margin: auto;
    padding: 0 settings.$m-size;
    max-width: 80rem;
}