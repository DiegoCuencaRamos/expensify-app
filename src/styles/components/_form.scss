@use '../base/settings';

.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: settings.$m-size;
    }
}

.form__error {
    margin: 0 0 settings.$m-size 0;
    font-size: italic;

}