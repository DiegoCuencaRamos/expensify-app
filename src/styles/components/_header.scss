@use '../base/settings';

.header {
    background-color: settings.$dark-blue;
}

.header__content {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: settings.$m-size 0;
}

.header__title {
    color: white;
    text-decoration: none;
    h1 {
        margin: 0;
    }
}