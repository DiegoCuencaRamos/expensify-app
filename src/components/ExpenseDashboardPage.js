import React from 'react';
import { ExpensesSumary } from './ExpensesSumary';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './EspenseListFilters';


const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSumary />
        <ExpenseList />
        <ExpenseListFilters />
    </div>
);

export default ExpenseDashboardPage;