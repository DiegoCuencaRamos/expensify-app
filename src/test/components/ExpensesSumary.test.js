import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSumary } from '../../components/ExpensesSumary';

test('should render ExpenseSumary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSumary expenseCount={1} expensesTotal={2350} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSumary with one expense or more', () => {
    const wrapper = shallow(<ExpensesSumary expenseCount={3} expensesTotal={10250} />);
    expect(wrapper).toMatchSnapshot();
});