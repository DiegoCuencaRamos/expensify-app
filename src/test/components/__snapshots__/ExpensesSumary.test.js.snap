// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSumary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    2,350 $
  </h1>
</div>
`;

exports[`should render ExpenseSumary with one expense or more 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    10,250 $
  </h1>
</div>
`;
